tests/test_postal_backend.py

```diff
@@ -0,0 +1,473 @@
+        self.assertEqual(msg.anymail_status.message_id, 1531)
+        self.assertEqual(msg.anymail_status.recipients['to1@example.com'].status, 'queued')
+        self.assertEqual(msg.anymail_status.recipients['to1@example.com'].message_id,
+                         1531)
+        self.assertEqual(msg.anymail_status.esp_response.content, response_content)
+
+    # noinspection PyUnresolvedReferences
+    def test_send_without_to_attaches_anymail_status(self):
+        """The anymail_status should be attached even if there are no `to` recipients"""
+        response_content = b"""{
+          "status": "success",
+          "time": 1.08,
+          "flags": {},
+          "data": {
+            "message_id": "9dfcc4df-09a6-4f1d-b535-0eb0a9f104a4@postal.example.com",
+            "messages": {
+              "cc@example.com": { "id": 1531, "token": "xLcafDRCVUFe" }
+            }
+          }
+        }"""
+        self.set_mock_response(raw=response_content)
+        msg = mail.EmailMessage('Subject', 'Message', 'from@example.com', cc=['cc@example.com'],)
+        sent = msg.send()
+        self.assertEqual(sent, 1)
+        self.assertEqual(msg.anymail_status.status, {'queued'})
+        self.assertEqual(msg.anymail_status.message_id, 1531)
+        self.assertEqual(msg.anymail_status.recipients['cc@example.com'].status, 'queued')
+        self.assertEqual(msg.anymail_status.recipients['cc@example.com'].message_id,
+                         1531)
+        self.assertEqual(msg.anymail_status.esp_response.content, response_content)
+
+    # noinspection PyUnresolvedReferences
+    def test_send_failed_anymail_status(self):
+        """ If the send fails, anymail_status should contain initial values"""
+        self.set_mock_response(status_code=500)
+        sent = self.message.send(fail_silently=True)
+        self.assertEqual(sent, 0)
+        self.assertIsNone(self.message.anymail_status.status)
+        self.assertIsNone(self.message.anymail_status.message_id)
+        self.assertEqual(self.message.anymail_status.recipients, {})
+        self.assertIsNone(self.message.anymail_status.esp_response)
+
+    # noinspection PyUnresolvedReferences
+    def test_send_unparsable_response(self):
+        """If the send succeeds, but a non-JSON API response, should raise an API exception"""
+        mock_response = self.set_mock_response(status_code=200,
+                                               raw=b"yikes, this isn't a real response")
+        with self.assertRaises(AnymailAPIError):
+            self.message.send()
+        self.assertIsNone(self.message.anymail_status.status)
+        self.assertIsNone(self.message.anymail_status.message_id)
+        self.assertEqual(self.message.anymail_status.recipients, {})
+        # TODO esp_response is None because post_to_esp calls raise_for_status which tries to deserialize the response
+        # self.assertEqual(self.message.anymail_status.esp_response, mock_response)
```

Comment:
This is an interesting one, I wonder what to do here.

In `raise_for_status`, called by `post_to_esp`, I need to deserialize the JSON response to figure out if the request was successful, because Postal doesn't really use HTTP status codes. If the response is unparsable, this would raise an error, not allowing `AnymailBaseBackend._send` to set `esp_response`:
https://github.com/anymail/django-anymail/blob/a7ea862ff147b8b9ce18786a194add199c1a2630/anymail/backends/base.py#L123-L125

I guess I could catch the deserialization error in `raise_for_status` and _not_ raise an error in this case. It will be raised later in `parse_recipient_status`.

Reply:
See my comment in the backend. I think the best fix is for the postal backend to *not* implement `raise_for_status`, and to move the error checking into `parse_recipient_status`.

Reply:
Done!